src/pkg/services/m365/api/teams.go

```diff
@@ -0,0 +1,165 @@
+package api
+
+import (
+	"context"
+
+	"github.com/alcionai/clues"
+	msgraphgocore "github.com/microsoftgraph/msgraph-sdk-go-core"
+	"github.com/microsoftgraph/msgraph-sdk-go/models"
+
+	"github.com/alcionai/corso/src/internal/common/str"
+	"github.com/alcionai/corso/src/internal/common/tform"
+	"github.com/alcionai/corso/src/internal/m365/graph"
+	"github.com/alcionai/corso/src/pkg/fault"
+	"github.com/alcionai/corso/src/pkg/logger"
+)
+
+const (
+	teamsAdditionalDataLabel    = "Team"
+	ResourceProvisioningOptions = "resourceProvisioningOptions"
+)
+
+// ---------------------------------------------------------------------------
+// controller
+// ---------------------------------------------------------------------------
+
+func (c Client) Teams() Teams {
+	return Teams{c}
+}
+
+// On creation of each Teams team a corrsponding group gets created.
+// The group acts as the protected resource, and all teams data like events,
+// drive and mail messages are owned by that group.
+
+// Teams is an interface-compliant provider of the client.
+type Teams struct {
+	Client
+}
+
+// GetAllTeams retrieves all groups.
+func (c Teams) GetAll(
```

Comment:
Apologies again on the design thrashing, but per our sync conversations, can you please change this package, file, and client backup to `Groups` instead of `Teams`?  That seems like it'll be a more appropriate starting point for the work we plan to implement.  If we move from there to needing a teams specific api, we can introduce another client for that.  Once the client naming is squared away, this is good to go.
```suggestion
func (c Groups) GetAll(
```

Reply:
done